### PR TITLE
onStepOut position fix

### PR DESCRIPTION
**Pull request by EvilHero90**
- now `onStepOut` returns `nextPosition` instead of `fromPosition` `onStepOut(creature, item, position, nextPosition)`
- `position` -> where the creature steps off
- `nextPosition` -> where the creature steps onto

### CODE REVIEW

**Comment by TheSumm** [comment]
`MoveEvents::onCreatureMove` is called after the creature has been moved to the new tile.
It should not be necessary to introduce `nextPosition`, since `creature->getPosition()` can be used.
By adding nextPosition/toPosition as the fourth parameter, it can be ensured that no scripts will break.

**Comment by wo11ven** [comment]
You mean lua Scripts that people have created or existing scripts in this repo? If we add a fifth variable, we will have a "dead" one that might cause confusion.

**Comment by EvilHero90** [comment]
I can only agree with what @wo11ven here just said it would cause more confusion.
Well, we have to agree that the way `onStepOut` is right now is not correct so it should be considered as a bug. It should have it's parameters fixed.
Vanilla files have been corrected, even tho we never used the `fromPosition` parameter somewhere anyway.
I don't see this really as a "breaking backwards" problem since you could say one of either parameters didn't return what the user expected it to be in the first place.

**Comment by TheSumm** [comment]
IMO it's not confusing at the moment. The parameters do exactly what they say right now.
`position` is the position of the tile/item which the script is assigned to.
`fromPosition` is the position where you came from.
Granted `fromPosition` is not very useful in case of `onStepOut`, since it is identical to `position`. Nevertheless, it is exactly what I would expect from it.

The misconception in #2586 is, that `position` is referring to the new position, which is does not and would not after any of these changes.
Therefore, I see no reason to break backwards compatibility in order to change something which works as intended/expected.
For now I would either leave it as it is, since you can already use `creature:getPosition()` to obtain the new position, OR introduce `toPosition` as a new parameter to both `onStepIn` and `onStepOut`.

Once we are willing to break backwards compatibility, it might be the best idea to get rid of `position` and just use `fromPosition` and `toPosition`.

**Comment by EvilHero90** [comment]
I don't mind about `fromPos` and `toPos` those where my first intentions anyway but as it would have caused a complete re work on all scripts with `onStepIn/onStepOut` I have refrained from it.
How about I'll make the changes according to `fromPos` and `toPos` and once 1.3 is released, we should merge this.

**Review by djarek — DISMISSED** [review]

File: src/movement.cpp [hunk 1]
```diff
@@ -940,8 +940,8 @@ bool MoveEvent::executeStep(Creature* creature, Item* item, const Position& pos)
 	LuaScriptInterface::pushUserdata<Creature>(L, creature);
 	LuaScriptInterface::setCreatureMetatable(L, -1, creature);
 	LuaScriptInterface::pushThing(L, item);
-	LuaScriptInterface::pushPosition(L, pos);
-	LuaScriptInterface::pushPosition(L, creature->getLastPosition());
+	LuaScriptInterface::pushPosition(L, eventType == MOVE_EVENT_STEP_OUT ? creature->getNextPos() : pos);
```
Comment (TheSumm) [inline src/movement.cpp]:
```suggestion
	LuaScriptInterface::pushPosition(L, eventType == MOVE_EVENT_STEP_OUT ? creature->getPosition() : pos);
```

File: src/creature.h [hunk 2]
```diff
@@ -507,6 +514,7 @@ class Creature : virtual public Thing
 		Outfit_t defaultOutfit;
 
 		Position lastPosition;
+		Position nextPosition;
```
Comment (TheSumm) [inline src/creature.h]: `nextPosition` is not needed

**Pull request closed without merging**